openapi-diff/src/core/OpenApiDiff.Core/Settings.cs

```diff
@@ -240,21 +244,37 @@ public void Validate()
                 var doc = property.GetCustomAttributes<SettingsInfoAttribute>().FirstOrDefault();
                 if (doc != null && doc.IsRequired && property.GetValue(this) == null)
                 {
-                    Console.WriteLine(String.Format(Resources.ParameterValueIsMissing, property.Name));
+                    Console.WriteLine(string.Format(Resources.ParameterValueIsMissing, property.Name));
                     throw new Exception(string.Format(Resources.ParameterValueIsMissing, property.Name));
                 }
             }
 
-            // Validate input Files
-            if (!File.Exists(Instance.OldSpec))
+            if (!ValidateUri(Instance.OldSpec))
             {
-                throw new Exception(String.Format(Resources.InputMustBeAFile, "-old"));
+                throw new Exception(string.Format(Resources.InputMustBeAFileOrUrl, "-old"));
             }
 
-            if (!File.Exists(Instance.NewSpec))
+            if (!ValidateUri(Instance.NewSpec))
             {
-                throw new Exception(String.Format(Resources.InputMustBeAFile, "-new"));
+                throw new Exception(string.Format(Resources.InputMustBeAFileOrUrl, "-new"));
             }
         }
+
+        private bool ValidateUri(Uri uri)
+        {
+            if (Instance.OldSpec.Scheme == Uri.UriSchemeFile &&
+                File.Exists(uri.AbsolutePath))
+            {
+                return true;
+            }
+
+            if (Instance.OldSpec.Scheme == Uri.UriSchemeHttp || Instance.OldSpec.Scheme == Uri.UriSchemeHttps)
+            {
+                // todo: validate the uri
```

Comment:
wouldn't a call to this do a basic validation ? https://msdn.microsoft.com/en-us/library/system.uri.iswellformeduristring(v=vs.110).aspx